core/org.wso2.carbon.base/src/test/java/org/wso2/carbon/base/ServerConfigurationTest.java

```diff
@@ -0,0 +1,185 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.internal.junit.ArrayAsserts.assertArrayEquals;
+
+/**
+ * Test class for ServerConfiguration related methods.
+ */
+public class ServerConfigurationTest {
+
+    @BeforeClass
+    public void createInstance() throws ServerConfigurationException, NoSuchFieldException,
+            IllegalAccessException {
+        assertFalse(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+        InputStream inputStream = readFile("carbon.xml");
+        ServerConfiguration.getInstance().forceInit(inputStream);
+        assertTrue(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+    }
+
+    @BeforeMethod
+    public void setIsInitializedToFalse() throws NoSuchFieldException, IllegalAccessException {
+        //setting isInitialized to false before every test
+        Field isInitializedField = ServerConfiguration.class.getDeclaredField("isInitialized");
+        isInitializedField.setAccessible(true);
+        isInitializedField.set(ServerConfiguration.getInstance(), false);
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testForceInitWithLocationOfXMLConfig() throws NoSuchFieldException, IllegalAccessException,
+            ServerConfigurationException {
+        assertFalse(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+        ServerConfiguration.getInstance().forceInit(CarbonBaseUtilsTest.class.getClassLoader().getResource("carbon" +
+                ".xml").toString());
+        assertTrue(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testForceInitWithXMLConfigAndBooleanAsArgs() throws NoSuchFieldException, IllegalAccessException,
+            ServerConfigurationException {
+        assertFalse(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+        ServerConfiguration.getInstance().forceInit(CarbonBaseUtilsTest.class.getClassLoader().getResource("carbon" +
+                ".xml").toString(), false);
+        assertTrue(getIsInitialized(ServerConfiguration.getInstance(), "isInitialized"));
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testSetConfigurationProperty() throws Exception {
+        ServerConfiguration.getInstance().setConfigurationProperty("ServerKey", "AM");
+        //need to update the test after issue #1560 is fixed
+    }
+
+    @Test(groups = {"org.wso2.carbon.base"})
+    public void testOverrideConfigurationProperty() {
+        ServerConfiguration.getInstance().overrideConfigurationProperty("RegistryHttpPort", "9764");
+        //need to update the test after issue #1560 is fixed
```

Comment:
Shall we include a TODO here.

Example: // TODO: Need to update test after issue #1560 is fixed